federation/pkg/federation-controller/util/federated_informer.go

```diff
@@ -146,9 +164,16 @@ func NewFederatedInformer(federationClient federation_release_1_4.Interface, tar
 					return
 				}
 				if isClusterReady(oldCluster) != isClusterReady(curCluster) || !reflect.DeepEqual(oldCluster.Spec, curCluster.Spec) {
+
+					if clusterLifecycle.ClusterUnavailable != nil {
+						clusterLifecycle.ClusterUnavailable(oldCluster, federatedInformer, federatedInformer.GetTargetStore())
+					}
 					federatedInformer.deleteCluster(oldCluster)
 					if isClusterReady(curCluster) {
 						federatedInformer.addCluster(curCluster)
+						if clusterLifecycle.ClusterAvailable != nil {
+							clusterLifecycle.ClusterAvailable(curCluster, federatedInformer, federatedInformer.GetTargetStore())
+						}
```

Comment:
With every cluster spec update we have to rebuild informer and execute factory function. The current behavior is in sync with what is done with informers. I'm aware that this is not perfect but currently it may be a lesser evil than having different behavior in factory and lifecycle handler.